### PR TITLE
feat(ui): pipeline UX improvements

### PR DESCRIPTION
**Pull request by Marvin9**
- default fit the whole pipeline into view
- drag like maps and zoom in/out using mouse wheel
- [x] remember zoom preference - https://github.com/akuity/kargo/issues/2775
- [x] Add screenshot/video

https://github.com/user-attachments/assets/14afcf3f-1686-4f36-bf70-1c06338a4cff

### CODE REVIEW

**Review by jessesuen — APPROVED** [review]
Not a code review, but approval based on behavior demonstrated in video

**Review by krancour — APPROVED** [review]
🔥 This is amazing and I also approve based on a review of the functionality.

idk if @rpelczar may wish to review implementation details as well since this appears non-trivial.

File: ui/src/utils/use-temporary.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const useTemporaryBoolean = (timeout = 500 /* milliseconds */) => {
+  const [init, setInit] = useState(false);
+
+  useEffect(() => {
+    if (init) {
+      setTimeout(() => {
+        setInit(false);
+      }, timeout);
+    }
+  }, [init]);
```
Comment (rpelczar) [inline ui/src/utils/use-temporary.ts:15]: Don't forget clear interval in the cleanup function.

**Review by rpelczar — APPROVED** [review]
Looks amazing! I left one comment with one small thing to improve.

**Comment by rpelczar** [comment]
I also think the save button is overkill. I like it, but not sure if it is as useful as the reset button.
This is my personal opinion, so you don't have to do anything with that.

**Comment by Marvin9** [comment]
I think that makes sense @rpelczar, we have reset view anyways so I have removed explicit save view preference and it autosaves now. Let me know if you think otherwise @jessesuen @krancour

**Comment by 02strich** [comment]
Btw, this breaks kargo on firefox due to https://developer.mozilla.org/en-US/docs/Web/API/Element/computedStyleMap not being supported there today

**Comment by hiddeco** [comment]
@02strich, thank you for bringing this to our attention. We've noticed this issue internally as well, and will work on a solution. Appreciate the report! 🙇

**Pull request merged**